static/app/components/devtoolbar/components/replay/replayPanel.tsx

```diff
@@ -0,0 +1,136 @@
+import {useEffect, useState} from 'react';
+import {css} from '@emotion/react';
+import type {ReplayRecordingMode} from '@sentry/types';
+
+import {Button} from 'sentry/components/button';
+import SentryAppLink from 'sentry/components/devtoolbar/components/sentryAppLink';
+import {resetFlexRowCss} from 'sentry/components/devtoolbar/styles/reset';
+import ProjectBadge from 'sentry/components/idBadge/projectBadge';
+import {IconPause, IconPlay} from 'sentry/icons';
+import type {PlatformKey} from 'sentry/types/project';
+
+import useConfiguration from '../../hooks/useConfiguration';
+import {panelInsetContentCss, panelSectionCss} from '../../styles/panel';
+import {smallCss} from '../../styles/typography';
+import PanelLayout from '../panelLayout';
+
+const TRUNC_ID_LENGTH = 16;
+const POLL_INTERVAL_MS = 3000;
+
+export default function ReplayPanel() {
+  const {projectSlug, projectId, projectPlatform, trackAnalytics, SentrySDK} =
+    useConfiguration();
+  const replay =
+    SentrySDK && 'getReplay' in SentrySDK ? SentrySDK.getReplay() : undefined;
+
+  // sessionId is undefined iff we are recording in session OR buffer mode.
+  const [sessionId, setSessionId] = useState<string | undefined>(() =>
+    replay?.getReplayId()
+  );
+  const [recordingMode, setRecordingMode] = useState<ReplayRecordingMode>(
+    () => replay?._replay.recordingMode
```

Comment:
Could create a helper to get this `_replay` value, that'll help so we only have to squash the lint warning in one place.

To start it'll be something like this: 
```
function useFoo() {
  const {SentrySDK} = useConfiguration();
  const replay =
    SentrySDK && 'getReplay' in SentrySDK ? SentrySDK.getReplay() : undefined;
  return replay?._replay;
```

But we'll probably actually need to return some other information in cases like:
- the user has too old of an SDK, idk what the problem is with 8.18, but v7 is also something they could have
- there is no replayIntegration at all and we want to show a message about that

Reply:
Could the helper accept the `replay` (replay integration) as an argument, instead of calling useConfiguration and getReplay every time?

Reply:
ya, i wasn't trying to be too prescriptive. 

The notes you made about 8.18 needing try/catch too could also factor in here, we should add those!

Reply:
Sounds good, 8.18 will not be disabled but some of the fx calls will probably throw, so I need to handle that..
The button won't be disabled unless the SDK is older than the `getReplay` function. Need to check which version it was released.

Reply:
Got it - just want to see which one it is for myself!

Reply:
I haven't found when `getReplay()`/the public API was first released. If we need the exact version, maybe @billyvg can help?